src/Altinn.App.Core/Features/Maskinporten/MaskinportenClient.cs

```diff
@@ -0,0 +1,273 @@
+using System.Diagnostics;
+using System.Text.Json;
+using Altinn.App.Core.Features.Maskinporten.Exceptions;
+using Altinn.App.Core.Features.Maskinporten.Models;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Altinn.App.Core.Features.Maskinporten;
+
+/// <inheritdoc/>
+public sealed class MaskinportenClient : IMaskinportenClient
+{
+    /// <summary>
+    /// The margin to take into consideration when determining if a token has expired (seconds).
+    /// <remarks>This value represents the worst-case latency scenario for <em>outbound</em> connections carrying the access token.</remarks>
+    /// </summary>
+    internal const int TokenExpirationMargin = 30;
```

Comment:
I have had this conversation with @martinothamar as well. I don't personally believe a relative value makes any sense, mainly because:

We added a margin here to account for outbound latency. The inbound latency between Maskinporten and the library code has already been accounted for. The outbound latency for authorized http calls aren't really related to a percentage of when the token expires. They are related to an absolute time when the token is considered expired. What we want to assure here, is that we don't ship off a token that expires in transit.

In my mind, this requires a fixed value and does not relate to the relative lifespan of the token. 30 seconds is obviously a generous buffer, but at the same time is lower than the default http client timeout (100 seconds): https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout?view=net-8.0